tests/Base/StringUtilityInterfaceTest.php

```diff
@@ -254,6 +268,43 @@ public function removeSuffixDataProvider()
 
 
 
+    public function substringDataProvider()
+    {
+        return [
+            'only unique words' => [
+                'string'            => 'a line with only unique words',
+                'after'             => 'with',
+                'before'            => 'unique',
+                'expectedResult'    => ' only '
+            ],
+            'duplicated words' => [
+                'string'            => 'a line with some duplicated words like line some or words',
+                'after'             => 'some',
+                'before'            => 'words',
+                'expectedResult'    => ' duplicated '
+            ],
+            'no after' => [
+                'string'            => 'no after available here',
+                'after'             => 'some',
+                'before'            => 'here',
+                'expectedResult'    => 'no after available '
+            ],
+            'no before' => [
+                'string'            => 'no before available here',
+                'after'             => 'before',
+                'before'            => 'some',
+                'expectedResult'    => ' available here'
+            ],
+            'no after and no before' => [
+                'string'            => 'no before and after available here',
+                'after'             => 'some',
+                'before'            => 'word',
+                'expectedResult'    => 'no before and after available here'
+            ]
```

Comment:
Magst du hier noch Testdaten für folgende Fälle hinzufügen:
- $after ist der leere String, $before in $string enthalten
- $before ist der leere String, $after in $string enthalten
- $after = $before = ''
- $after befindet sich hinter $before